File: 789_range_of_number.py
def b_search(ls, number, n):
    i = 0
    j = n - 1
    while i < j:
        mid = i + j >> 1
        if ls[mid] >= number:
            j = mid
        else:
            i = mid + 1
    if ls[i] != number:
        return -1, -1
    else:
        l = i
        i = 0
        j = n - 1
        while i < j:
            mid = i + j + 1 >> 1
            if ls[mid] <= number:
                i = mid
            else:
                j = mid - 1
        r = i
    return l, r


[n, q] = [int(x) for x in input().split()]
ls = [int(x) for x in input().split()]
for i in range(q):
    number = int(input())
    l, r = b_search(ls, number, n)
    print(l, r)

File: cpp_solution/section_1/789_range_of_number.cpp
#include <iostream>

using namespace std;

const int N = 100010;

int a[N];

int main() {
    int n, q;
    scanf("%d%d", &n, &q);
    for (int i = 0; i < n; i++) {
        scanf("%d", &a[i]);
    }
    while(q--) {
        int number;
        scanf("%d", &number);
        int l = 0, r = n - 1, mid = 0;
        while(l < r) {
            mid = l + r >> 1;
            if(a[mid] >= number) {
                r = mid;
            }
            else {
                l = mid + 1;
            }
        }
        if (a[l] != number) {
            cout << "-1 -1" << endl;
        }
        else {
            cout << l << " ";
            l = 0;
            r = n - 1;
            while(l < r) {
                mid = l + r + 1 >> 1;
                if(a[mid] <= number) {
                    l = mid;
                }
                else {
                    r = mid - 1;
                }
            }
            cout << l << endl;
        }

    }
    return 0;
}